src/compiler/scanner.ts

```diff
@@ -559,6 +562,9 @@ namespace ts {
 
         // Keep in sync with couldStartTrivia
         while (true) {
+            if (pos >= text.length) {

```

Comment:
Consider making the loop `while (pos < text.length)` (maybe even locally capturing `text.length`).

Reply:
I did it this way because the area after the loop is currently unreachable and changing that seemed less readable.

Does capturing `text.length` make a different in JS?  I believe the .net vm detects the pattern is does smart things and I assumed v8 did the same.